crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -498,9 +498,9 @@ pub fn check_member_access(
 
     if let Type::Struct(_, s) = &lhs_type {
         let s = s.borrow();
-        if let Some(field) = s.fields.iter().find(|(name, _)| name == &access.rhs) {
+        if let Some(field) = s.get_field(&access.rhs.0.contents) {
             // TODO: Should the struct's visibility be applied to the field?
```

Comment:
This is a good question. Can we open an issue for this?

Reply:
Yeah I think you are correct.

> This would force every use of the struct to have the same field visibilities though.

This is fine in my opinion, its the same as in Rust where the visibility is defined once when you define the struct